### PR TITLE
[dotnet] Add support for the static registrar.

### PR DESCRIPTION
**Pull request by rolfbjarne**

### CODE REVIEW

File: tools/common/Application.cs [hunk 1]
```diff
@@ -899,6 +899,42 @@ public void ParseAbi (string abi)
...
+				break;
+			case "dynamic":
+				Registrar = RegistrarMode.Dynamic;
+				break;
+			case "default":
+				Registrar = RegistrarMode.Default;
+				break;
+			case "partial":
+			case "partial-static":
+				Registrar = RegistrarMode.PartialStatic;
```
Comment (spouliot) [inline tools/common/Application.cs:919]: what will happen inside `mtouch` ?

**Review by spouliot — APPROVED** [review]

**Review by chamons — APPROVED** [review]

**Review by whitneyschmidt — APPROVED** [review]

File: dotnet/targets/Xamarin.Shared.Sdk.targets [hunk 2]
```diff
@@ -328,24 +333,29 @@
...
+			<_CompileNativeExecutableFile Include="@(_RegistrarFile)">
 				<OutputFile>$(_IntermediateNativeLibraryDir)%(Filename).o</OutputFile>
-			</_MainFile>
+			</_CompileNativeExecutableFile>
 			<_XamarinMainIncludeDirectory Include="$(_XamarinIncludeDirectory)" />
 		</ItemGroup>
+	</Target>
+
+	<Target Name="_CompileNativeExecutable"
+		DependsOnTargets="_DetectSdkLocations;_ComputeTargetArchitectures;_GenerateBundleName;_ComputeVariables;_GetMinimumOSVersion;_ComputeNativeExecutableInputs"
```
[the remark this answers is not in the document]
Comment (rolfbjarne) [reply dotnet/targets/Xamarin.Shared.Sdk.targets:349]: Good point, and the answer is no, it's not needed.

**Comment by rolfbjarne** [comment]
Test failure is unrelated:

* xammac tests: https://github.com/xamarin/maccore/issues/2160

File: tools/common/Application.cs (hunk 1 above)
Comment (rolfbjarne) [reply tools/common/Application.cs:920]: I don't know, so I added an ifdef to avoid the problem.

**Pull request merged**